crates/configuration/src/network.rs

```diff
@@ -0,0 +1,260 @@
+use std::marker::PhantomData;
+
+use crate::{
+    global_settings::{GlobalSettings, GlobalSettingsBuilder},
+    hrmp_channel::{self, HrmpChannelConfig, HrmpChannelConfigBuilder},
+    parachain::{self, ParachainConfig, ParachainConfigBuilder},
+    relaychain::{self, RelaychainConfig, RelaychainConfigBuilder},
+    shared::macros::states,
+};
+
+/// A network configuration, composed of a relaychain, parachains and HRMP channels.
+#[derive(Debug, Clone, PartialEq)]
+pub struct NetworkConfig {
+    // The global settings applied to the network.
+    global_settings: GlobalSettings,
+
+    /// Relaychain configuration.
+    relaychain: Option<RelaychainConfig>,
```

Comment:
I understand that the builder will `drive` the users to set the `relaychain` here, but I think we can drop the inner config and make two `similar` structs for `NetworkConfig` / `NetworkConfigBuider` making `optional` in the latest. wdyt?

Reply:
Resolved, since we already talked.